src/plugins/dashboard/public/application/actions/copy_to_dashboard_modal.tsx

```diff
@@ -0,0 +1,138 @@
+  embeddable: IEmbeddable;
+  dashboardId?: string;
+  closeModal: () => void;
+}
+
+export function CopyToDashboardModal({
+  PresentationUtilContext,
+  stateTransfer,
+  capabilities,
+  dashboardId,
+  embeddable,
+  closeModal,
+}: CopyToDashboardModalProps) {
+  const [dashboardOption, setDashboardOption] = useState<'new' | 'existing'>('existing');
+  const [selectedDashboard, setSelectedDashboard] = useState<{ id: string; name: string } | null>(
+    null
+  );
+
+  const onSubmit = useCallback(() => {
+    const state = {
+      input: omit(embeddable.getInput(), 'id'),
+      type: embeddable.type,
+    };
+
+    const path =
+      dashboardOption === 'existing' && selectedDashboard
+        ? `#${createDashboardEditUrl(selectedDashboard.id, true)}`
+        : `#${DashboardConstants.CREATE_NEW_DASHBOARD_URL}`;
+
+    closeModal();
+    stateTransfer.navigateToWithEmbeddablePackage('dashboards', {
+      state,
+      path,
+    });
+  }, [dashboardOption, embeddable, selectedDashboard, stateTransfer, closeModal]);
+
+  return (
+    <PresentationUtilContext>
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>{dashboardCopyToDashboardAction.getDisplayName()}</EuiModalHeaderTitle>
+      </EuiModalHeader>
+
+      <EuiModalBody>
+        <>
+          <EuiText className="dshCopyToParagraph">
+            <p>{dashboardCopyToDashboardAction.getDescription()}</p>
+          </EuiText>
+          <EuiFormRow hasChildLabel={false}>
+            <EuiPanel color="subdued" hasShadow={false} data-test-subj="add-to-dashboard-options">
+              <div>
+                {capabilities.canEditExisting && (
+                  <>
+                    <EuiRadio
+                      checked={dashboardOption === 'existing'}
+                      data-test-subj="add-to-existing-dashboard-option"
+                      id="existing-dashboard-option"
+                      name="dashboard-option"
+                      label={dashboardCopyToDashboardAction.getExistingDashboardOption()}
+                      onChange={() => setDashboardOption('existing')}
+                    />
+                    <div className="savAddDashboard__searchDashboards">
+                      <DashboardPicker
+                        isDisabled={dashboardOption !== 'existing'}
+                        idsToOmit={dashboardId ? [dashboardId] : undefined}
+                        onChange={(dashboard) => setSelectedDashboard(dashboard)}
+                      />
+                    </div>
+                    <EuiSpacer size="s" />
+                  </>
+                )}
+                {capabilities.canCreateNew && (
+                  <>
+                    <EuiRadio
+                      checked={dashboardOption === 'new'}
+                      data-test-subj="add-to-new-dashboard-option"
+                      id="new-dashboard-option"
+                      name="dashboard-option"
+                      disabled={!dashboardId}
```

Comment:
Close! You can't copy from an unsaved dashboard, to another unsaved dashboard. This is because the URLs are the same, it would just append a copy of the panel.